scribejava-apis/src/main/java/com/github/scribejava/apis/SalesforceApi.java

```diff
@@ -1,28 +1,39 @@
 package com.github.scribejava.apis;
 
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+
 import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocket;
 
 import com.github.scribejava.apis.salesforce.SalesforceJsonTokenExtractor;
 import com.github.scribejava.core.builder.api.DefaultApi20;
 import com.github.scribejava.core.extractors.TokenExtractor;
 import com.github.scribejava.core.model.OAuth2AccessToken;
 import com.github.scribejava.core.model.Verb;
-import java.io.IOException;
-import java.security.KeyManagementException;
-import java.security.NoSuchAlgorithmException;
-import javax.net.ssl.SSLSocket;
 
 /**
  * This class is an implementation of the Salesforce OAuth2 API.
  */
 public class SalesforceApi extends DefaultApi20 {
 
-    private static final String BASE_URL = "https://login.salesforce.com/services/oauth2";
-    private static final String ACCESS_URL = BASE_URL + "/token";
-    private static final String AUTHORIZE_URL = BASE_URL + "/authorize";
+    private String hostName = "";
```

Comment:
It should not be non-final.
Imaging this code
```
final SalesforceApi api = SalesforceApi.instance();
final ServiceBuilder serviceBuilder = ...
    .build(api);
...
service.getAuthorizationUrl();
...
api.withSandbox();
service.getAccessToken(codeEncoded);
```

It should looks like this
```
private final String hostName;
protected SalesforceApi(String hostName) {
    this.hostName = hostName;
    ....
}

private static class ProdInstanceHolder {
    private static final SalesforceApi INSTANCE = new SalesforceApi(PRODUCTION_HOST);
}

private static class SandboxInstanceHolder {
    private static final SalesforceApi INSTANCE = new SalesforceApi(SANDBOX_HOST);
}

public static SalesforceApi instance() {
    return ProdInstanceHolder.INSTANCE;
}

public static SalesforceApi sandboxInstance() {
    return SandboxInstanceHolder.INSTANCE;
}
```

although singletons for sandbox is overkill. Let it be just
```
private final String hostName;
protected SalesforceApi(String hostName) {
    this.hostName = hostName;
    ....
}

private static class InstanceHolder {
    private static final SalesforceApi INSTANCE = new SalesforceApi(PRODUCTION_HOST);
}

public static SalesforceApi instance() {
    return InstanceHolder.INSTANCE;
}

public static SalesforceApi sandbox(SANDBOX_HOST) {
    return new SalesforceApi();
}
```